### PR TITLE
[MPOM-440] Add maven-checkstyle-plugin to pluginManagement

### PR DESCRIPTION
**Pull request by slawekjaranowski**

### CODE REVIEW

**Review by slachiewicz — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -181,6 +182,11 @@ under the License.
           <artifactId>maven-clean-plugin</artifactId>
           <version>${version.maven-clean-plugin}</version>
         </plugin>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-checkstyle-plugin</artifactId>
+          <version>${version.maven-checkstyle-plugin}</version>
+        </plugin>
```
Comment (ctubbsii) [inline pom.xml:189]: The only problem with this is that the maven-checkstyle-plugin often uses a very old version of checkstyle, so to be useful to users, they still often have to define the plugin to override the dependency, as in:

```xml
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-checkstyle-plugin</artifactId>
        <dependencies>
          <dependency>
            <groupId>com.puppycrawl.tools</groupId>
            <artifactId>checkstyle</artifactId>
            <version>10.12.2</version>
          </dependency>
        </dependencies>
        <executions>
          <execution>
            <id>check-style</id>
            <goals>
              <goal>check</goal>
            </goals>
            <configuration>
              <!-- execution configuration here -->
            </configuration>
          </execution>
        </executions>
      </plugin>
```

Comment (slawekjaranowski) [reply pom.xml:189]: yes it is true.
But newer checkstyle required JDK 11+, so in this place we can not make such assumptions.
child project also define specific plugin configurations.
So here only plugin version is reasonable.

Comment (slachiewicz) [reply pom.xml:189]: Maybe we can define checkstyle version as additional property?

Comment (slawekjaranowski) [reply pom.xml:189]: I don't want to much complicate the ASF parent pom.
So I think that managing plugin version is good enough.

**Comment by jira-importer** [comment]
Resolve #385

**Pull request merged**